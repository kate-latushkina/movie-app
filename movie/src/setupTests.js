// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
