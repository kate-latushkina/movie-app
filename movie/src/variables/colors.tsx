const colors = {
  darkViolet: "#593158",
  darkGrey: "#424242",
  lightGrey: "#555555",
  pink: "#f65261",
  white: "#fff",
};

export default colors;
