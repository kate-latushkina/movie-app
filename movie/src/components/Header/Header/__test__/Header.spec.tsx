import React from "react";
import renderer from "react-test-renderer";
import HeaderColored from "../Header";
import "../../../../setupTests";
import colors from "../../../../variables/colors";

describe("should render Header component", () => {
  // it("should contain HeaderColored", () => {
  //   const wrapper = renderer.create(<HeaderColored />).toJSON();
  //   expect(wrapper).toHaveStyleRule("background-color", colors.darkViolet);
  // });

  it("should render Header snapshot", () => {
    const wrapper = renderer.create(<HeaderColored />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
