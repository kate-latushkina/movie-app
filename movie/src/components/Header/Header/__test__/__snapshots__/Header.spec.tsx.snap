// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component should render Header snapshot 1`] = `
.c4 {
  width: 100%;
  height: 40px;
}

.c5 {
  width: 70%;
  height: 100%;
  border-radius: 2px;
  border: none;
}

.c1 {
  color: #f65261;
}

.c3 {
  text-transform: uppercase;
  font-weight: 100;
  color: #fff;
}

.c2 {
  padding: 0 10%;
}

.c0 {
  width: 100%;
  background-color: #593158;
  padding: 20px 15% 100px;
  box-sizing: border-box;
  border-bottom: 10px solid #555555;
}

<div
  className="MuiGrid-root c0"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <div
      className="MuiGrid-root c1"
    >
      <b>
        netflix
      </b>
      <span>
        roulette
      </span>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        + Add movie
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root c2"
  >
    <h2
      className="c3"
    >
      Find your movie
    </h2>
    <div
      className="MuiGrid-root c4 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <input
        className="c5 search-panel_input"
        placeholder="What do you want to watch?"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained search-panel_button MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</div>
`;
